File: concurrency/sumofprimes.exs
# Sum of primes
defmodule Parallelism do

  def is_prime(n) when n <= 1, do: false
  def is_prime(2), do: true

  def is_prime(n) do
    limit = :math.sqrt(n) |> :math.ceil() |> trunc()
    not Enum.any?(2..limit, fn i -> rem(n, i) == 0 end)
  end


  def sum_of_primes(n) do
    1..n
    |> Enum.filter(&is_prime/1)
    |> Enum.sum()
  end


  def sum_of_primes_recursive(n) do
    sum_of_primes_recursive(1, n, 0)
  end

  defp sum_of_primes_recursive(i, n, sum) when i > n, do: sum

  defp sum_of_primes_recursive(i, n, sum) do
    if is_prime(i) do
      sum_of_primes_recursive(i + 1, n, sum + i)
    else
      sum_of_primes_recursive(i + 1, n, sum)
    end
  end

end
IO.write("Enter number: ")
input = IO.read(:line) |> String.trim() |> String.to_integer()
result = Parallelism.sum_of_primes(input)
IO.puts("Sum of primes: #{result}")

File: concurrency/sumofprimes.cpp
#include <iostream>
#include <cmath>
using namespace std;



bool isPrime(unsigned long long n){
    //cout << "Checking if " << n << " is prime" << endl;
    unsigned long long b = ceil(sqrt(n));
    //cout << "b: " << b << endl;
    if (n <= 1) return false;
    if (n == 2) return true; // 2 is prime
    for (int i = 2; i <= b; i++) {
        if (n % i == 0) {
            return false;
        }
    }
    return true;
    
}
unsigned long long sumofprimes(unsigned long n){
    unsigned long long sum = 0;
    for (unsigned long long i = 1; i <= n; i++) {
        //cout << i << " " << isPrime(i) << endl;
        if (isPrime(i)) {
            //cout << i << " is prime" << endl;
            sum += i;
        }
    }
    return sum;
}

int main(int argc, char *argv[]) {
    unsigned long primeNumber = atoi(argv[1]);
    cout << "Sum of primes: " << sumofprimes(primeNumber) << endl;


    return 0;
}